packages/now-routing-utils/test/superstatic.spec.js

```diff
@@ -168,6 +168,8 @@ test('convertRedirects', () => {
       source: '/feedback/((?!general).*)',
       destination: '/feedback/general',
     },
+    { source: '/firebase/:id', destination: 'https://$1.firebase.com/' },
+    { source: '/firebase/:id', destination: 'https://$1.firebase.com:8080/' },
```

Comment:
This is wrong. You can't mix `:id` syntax with `$1` syntax.

Its either a regex and $1 or a path segment with another path segment

Reply:
Right, updated to only use regex for the tests instead of mixing them

Reply:
What I mean is that there are two ways for users to define source/destination pairs.

1. They use the path segment matching `:id` in the source and path segment matching `:id` in the `destination` (optional). If the path segment is missing from the destination, then it is assumed to be a query string parameter.
2. They use regex groups `(.*)` in the source and `$1` numbered matches in the `destination`. This does add query string parameters since the matched groups are unnamed.

Your above test mixed both which will cause problems because the former is based on name and the latter is based on numbered index. So even though it works for this one case, I don't think it will work for the general case.

Consider this example:

```json
{ source: '/(new|edit)/:id', destination: 'https://$1.firebase.com' },
```
What should happen? Does `$1` represent `edit` or does it represent `id`? Should the query string parameter be added? Similar, what it `$2` was used. It gets really confusing and I don't think we should mix these two.

Reply:
I definitely see what you mean, looks like I missed updating the redirects tests above and only updated the rewrites ones